solidity/test/staker-rewards/MerkleDistributorTest.js

```diff
@@ -0,0 +1,217 @@
+const {createSnapshot, restoreSnapshot} = require("../helpers/snapshot")
+const {testValues} = require("./rewardsData.js")
+
+const ECDSARewardsDistributor = contract.fromArtifact("ECDSARewardsDistributor")
+const KeepToken = contract.fromArtifact("KeepToken")
+
+const BN = web3.utils.BN
+const chai = require("chai")
+chai.use(require("bn-chai")(BN))
+const expect = chai.expect
+
+describe("MerkleDistributor", () => {
+  let keepToken
+  let rewardsDistributor
+
+  before(async () => {
+    keepToken = await KeepToken.new()
+    rewardsDistributor = await ECDSARewardsDistributor.new(keepToken.address)
+  })
+
+  beforeEach(async () => {
+    await createSnapshot()
+  })
+
+  afterEach(async () => {
+    await restoreSnapshot()
+  })
+
+  describe("allocating rewards", () => {
+    it("should successfuly allocate KEEP tokens", async () => {
+      let balance = await keepToken.balanceOf(rewardsDistributor.address)
+      expect(balance).to.eq.BN(0)
+
+      await allocateTokens(
+        testValues.interval0.merkleRoot,
+        testValues.interval0.amountToAllocate
+      )
+
+      balance = await keepToken.balanceOf(rewardsDistributor.address)
+      expect(balance).to.eq.BN(testValues.interval0.amountToAllocate)
+    })
+
+    it("should successfuly emit RewardsAllocated event after rewards were allocated", async () => {
+      const merkleRoot = testValues.interval0.merkleRoot
+      const value = testValues.interval0.amountToAllocate
+
+      await keepToken.approve(rewardsDistributor.address, value)
+      const receipt = await rewardsDistributor.allocate(merkleRoot, value)
+
+      const amount = new BN(value)
+      expectEvent(receipt, "RewardsAllocated", {
+        merkleRoot,
+        amount,
+      })
+    })
+
+    it("should fail allocating KEEP tokens without prior approval", async () => {
+      await expectRevert(
+        rewardsDistributor.allocate(
+          testValues.interval0.merkleRoot,
+          testValues.interval0.amountToAllocate
+        ),
+        "SafeERC20: low-level call failed"
+      )
+    })
+  })
+
+  describe("claiming rewards", () => {
+    const merkle0 = testValues.interval0
+    const merkle1 = testValues.interval1
+    beforeEach(async () => {
+      await allocateTokens(
+        testValues.interval0.merkleRoot,
+        testValues.interval0.amountToAllocate
+      )
+
+      await allocateTokens(
+        testValues.interval1.merkleRoot,
+        testValues.interval1.amountToAllocate
+      )
+    })
+
+    it("should successfuly claim rewards and emit an event", async () => {
+      const merkleRoot = merkle0.merkleRoot
+      const index = merkle0.claims[0].index
+      const account = merkle0.claims[0].account
+      const amount = merkle0.claims[0].amount
+      const proof = merkle0.claims[0].proof
+
+      const claimed = await rewardsDistributor.claim(
+        merkleRoot,
+        index,
+        account,
+        amount,
+        proof
+      )
+
```

Comment:
Can we add `expect(await keepToken.balanceOf(account)).to.eq.BN(amount)` here as well? We have it in the loop for multiple rewards case, but having this check covered here would be also good so that we have all the assertions covered in this test.

Reply:
Done.